### ISSUE TITLE
Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://verify.walletconnect.com')

### ISSUE THREAD

**Issue by gritaliana** [WalletConnect/walletconnect-monorepo#3492]
**Describe the bug**
Connection with any wallet is not working on **https:** protocol.
I have tried Metamask, Trust wallet, your https://react-wallet.walletconnect.com/walletconnect. 
When modal is opened there is this error in the console, and scanning QR code is doing nothing.

<img width="920" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/9944499/87c95653-1c20-48a2-9b53-7d68ae1cb9be">

On **localhost** everything is working just fine!

**SDK Version (if relevant)**
"next": "^12.2.2",
"@walletconnect/modal": "^2.6.1",
"@walletconnect/sign-client": "^2.9.2",

**To Reproduce**
Client and modal are created and connected as it is described in docs, nothing unusual:

```
const signClient = await SignClient.init({
  projectId: DEFAULT_PROJECT_ID,
  relayUrl: DEFAULT_RELAY_URL, // wss://relay.walletconnect.com
})
const web3Modal = new WalletConnectModal({
  projectId: DEFAULT_PROJECT_ID,
  standaloneChains: ['eip155:1'],
})
```

On local environment everything works fine. We can connect, sign in, send transactions. With Metamask and Trust.

After deploying it to the **dev** environment, this error occurred every time I open the Modal.
In `https://cloud.walletconnect.com/` in our project for dev environment I have added to the **Origin allow list** section our domain for the dev. Domain is also verified.

<img width="808" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/9944499/8e4518e9-44cd-43c7-acfd-96ea0c0d6e22">

I have tried to connect from our **dev** by copying the QR code to the https://react-wallet.walletconnect.com/walletconnect and also got an error:

<img width="788" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/9944499/6c8a6cf5-be16-499f-838b-dbb5791d907e">

**Expected behavior**
Connection between our web dApp and any wallet, at least Metamask, is working.

**Desktop:**
 - OS: [macOS, iOS]
 - Browser [Chrome]
 - Version [Version 115.0.5790.170 (Official Build) (arm64)]

**Smartphone:**
 - Device: [iPhoneX]
 - OS: [iOS 16.5.1]

**Comment by ganchoradkov** [comment]
Hey @gritaliana we've just released `2.10.0` and this issue should be resolved. 
Please update and try again. I'm closing it and feel free to reopen if the issue persists.

**Comment by gritaliana** [comment]
Hi, thanks for the quick response! I have updated to the 2.10.0, but there is still an issue with verify.
On localhost still works fine.
On dev: Modal opens with no console Errors, but after I scan a QR code with Metamask wallet this error occurred and nothing else happens. Modal remains open, in wallet no response. Just an error in our web dApp:

<img width="772" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/9944499/85832766-eb26-47f8-83a4-123caecddc8f">

P.S. I don't see a reopen issue button anywhere here. So I will open a new one then.
